frontend/src/metabase/containers/Unsubscribe.tsx

```diff
@@ -0,0 +1,242 @@
+import { useState } from "react";
+import { useAsync } from "react-use";
+import { Location } from "history";
+import { t, jt } from "ttag";
+
+import { Center, Stack, Text } from "metabase/ui";
+import { SessionApi } from "metabase/services";
+import { isEmpty } from "metabase/lib/validate";
+import { NotFound } from "metabase/containers/ErrorPages";
+import { color } from "metabase/lib/colors";
+
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+import ExternalLink from "metabase/core/components/ExternalLink";
+import Button from "metabase/core/components/Button";
+import LogoIcon from "metabase/components/LogoIcon";
+import {
+  StyledMetabotLogo,
+  LayoutBody,
+  LayoutCard,
+  LayoutIllustration,
+  LayoutRoot,
+  CheckmarkIcon,
+} from "metabase/containers/Unsubscribe.styled";
+
+const ERRORS = {
+  MISSING_REQUIRED_PARAMETERS: "missing required parameters",
+};
+
+const SUBSCRIPTION = {
```

Comment:
Nit: There is a slightly more typesafe way of writing this. Right now the values are treated as string types, so I can call `setSubscriptionChange("any string value")` without a type error.

If we use `as const` and create a `Subscription` type from the object, then we can use that type to ensure the state value can only be `unsubscribe` or `resubscribe`. Same goes for the `ERRORS` object as well.

```typescript
const SUBSCRIPTION = {
  UNSUBSCRIBE: "unsubscribe",
  RESUBSCRIBE: "resubscribe",
} as const;
type Subscription = typeof SUBSCRIPTION[keyof typeof SUBSCRIPTION];
...
  const [subscriptionChange, setSubscriptionChange] = useState<Subscription>(
    SUBSCRIPTION.UNSUBSCRIBE,
  );
  ```

Reply:
Thank you very much for this - very neat! :heart: